### PR TITLE
House requests

### PR DESCRIPTION
**Pull request by billyeh**
Quite a bit of code cleanup/de-duplication as best as I could. Reusing as much code as I could with everything remaining extensible.

House requests pretty much as described in #438

### CODE REVIEW

File: ap/web_access/views.py [hunk 1]
```diff
@@ -74,13 +62,17 @@ class TAWebAccessUpdate(GroupRequiredMixin, generic.UpdateView):
     group_required = ['administration']
     raise_exception = True
 
+    def get_context_data(self, **kwargs):
```
Comment (billyeh) [inline ap/web_access/views.py]: Can get rid of this

File: ap/web_access/templates/web_access/ta_web_access_update.html [hunk 2]
```diff
@@ -1,26 +1 @@
...
-            {% buttons %}
-                <button type="submit" class="btn btn-primary">
-                    Submit
-                </button>
-            {% endbuttons %}
-        </form>
-    </div>
-
-{% endblock %}
+{% extends "requests/ta_comments.html" %}
```
Comment (billyeh) [inline ap/web_access/templates/web_access/ta_web_access_update.html]: Get rid of this

File: ap/templates/requests/request_list.html [hunk 3]
```diff
@@ -1,94 +1,103 @@
...
 
       <div class="panel-body">
-        {% include detail_template %}
+        {% include request.get_detail_template %}
       </div>
     {% endfor %}
     </div>
   {% empty %}
-    <h4><i>You have no {{ item_name }} requests.</i></h4>
+    <h4><i>You have no {{ view.model.model_verbose_name }}s.</i></h4>
```
Comment (billyeh) [inline ap/templates/requests/request_list.html]: Thanks, nice catch!

File: ap/house_requests/templates/request_list/linens_ta_buttons.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<a href="{% url 'house_requests:linens-modify-status' 'C' request.id %}">
```
Comment (billyeh) [inline ap/house_requests/templates/request_list/linens_ta_buttons.html]: Yeah, sure.

File: ap/house_requests/views.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  template_name = 'ta_comment.html'
+
+class FramingRequestTAComment(generic.UpdateView):
+  model = FramingRequest
+  fields = ['TA_comments']
+  template_name = 'ta_comment.html'
+
+class MaintenanceRequestDelete(generic.DeleteView):
+  model = MaintenanceRequest
+  success_url = reverse_lazy('house_requests:house_requests')
```
Comment (jontey) [inline ap/house_requests/views.py]: should this go to house_requests:maintenance-list?

File: ap/house_requests/views.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+class LinensRequestDelete(generic.DeleteView):
+  model = LinensRequest
+  success_url = reverse_lazy('house_requests:house_requests')
```
Comment (jontey) [inline ap/house_requests/views.py]: same here

File: ap/house_requests/views.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+class FramingRequestDelete(generic.DeleteView):
+  model = FramingRequest
+  success_url = reverse_lazy('house_requests:house_requests')
```
Comment (jontey) [inline ap/house_requests/views.py]: and here

File: ap/house_requests/urls.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+  url(r'^framing_list/$', views.FramingRequestList.as_view(), name='framing-list'),
+  url(r'^framing_request/$', views.FramingRequestCreate.as_view(), name='framing-request'),
+  url(r'^framing_update/(?P<pk>\d+)$', views.FramingRequestUpdate.as_view(), name='framing-update'),
+  url(r'^framing_detail/(?P<pk>\d+)$', views.FramingRequestDetail.as_view(), name='framing-detail'),
+  url(r'^framing_delete/(?P<pk>\d+)$', views.FramingRequestDelete.as_view(), name='framing-delete'),
+  url(r'^framing_modify/(?P<status>[CPF])/(?P<id>\d+)$', views.modify_framing_status, name='framing-modify-status'),
+  url(r'^framing_tacomment/(?P<pk>\d+)$', views.FramingRequestTAComment.as_view(), name='framing-tacomment'),
+
+  url(r'^house_requests/$', views.NewRequestPage, name='house_requests'),
```
Comment (jontey) [inline ap/house_requests/urls.py]: should we standardize the naming convention here? You have house_requests here but dashes for the other urls.

File: ap/house_requests/views.py (hunk 5 above)
Comment (billyeh) [inline ap/house_requests/views.py]: Yeah, you're right, that'd be better.

**Comment by jontey** [comment]
Looks good. Merging!

**Pull request merged**